### PR TITLE
Preview code docs

### PR DESCRIPTION
**Pull request by wjohnsto**
This PR adds JSDocs to all the functions and interfaces for the npm package.

### CODE REVIEW

**Comment by mindctrl** [comment]
@wjohnsto is this meant for `main` or `canary`?

**Comment by wjohnsto** [comment]
Clay instructed me to publish the PR to canary. I went through all the conflicts and resolved them.

**Comment by mindctrl** [comment]
Gotcha. It looks like this removes a lot of the other work we did. Is that intentional?

**Comment by wjohnsto** [comment]
Not intentional. Whenever does the merge is going to have to cherry pick stuff to include. If there's a specific way I can do this I'm all ears, this is just what happened when I tried to merge...

File: docs/previews/README.md [hunk 1]
```diff
@@ -110,120 +114,120 @@ export default function App({
...
 
 #### Configuration
 
 We need to let the frontend know about our WordPress instance. The framework expects a few environment variables with this information. Create a file in the root of the project `/.env.local`.
 
-```
+```bash
 # Base URL for WordPress
-NEXT_PUBLIC_WORDPRESS_URL=http://developer-hub.local
+NEXT_PUBLIC_WORDPRESS_URL=http://yourwp.com
```
Comment (nickcernis) [inline docs/previews/README.md:217]: http://yourwp.com resolves to the Washington Post.

Maybe use example.com here instead?

```suggestion
NEXT_PUBLIC_WORDPRESS_URL=https://example.com
```

File: examples/preview/README.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ Install the latest plugin for this project and [WPGraphQL](https://wordpress.org
 
 The framework expects a few environment variables. Create a file in the root of the example project `/.env.local`.
 
-```
+```bash
 # Base URL for WordPress
 NEXT_PUBLIC_WORDPRESS_URL=http://yourwpsite.com
```
Comment (nickcernis) [inline examples/preview/README.md:17]: We should probably update this one too. Not blocking and could be a separate PR.

```suggestion
NEXT_PUBLIC_WORDPRESS_URL=https://example.com
```

File: packages/headless/src/auth/middleware.ts [hunk 3]
```diff
@@ -3,6 +3,16 @@ import { authorize } from './authorize';
 import { storeAccessToken } from './cookie';
 import { getQueryParam } from '../utils';
 
+/**
+ * A Node handler for processing incomming requests to exchange an Authorization Code
```
Comment (nickcernis) [inline packages/headless/src/auth/middleware.ts:7]: Tiny typo in 'incomming':

```suggestion
 * A Node handler for processing incoming requests to exchange an Authorization Code
```
